### PR TITLE
Dictionary

### PR DESCRIPTION
**Pull request by ginge**
add dictionary and tuple support

### CODE REVIEW

File: Utilities/runtests.py [hunk 1]
```diff
@@ -27,11 +27,12 @@
 plat = ThisPlatform(qemu = args.qemu[0])
 
 print("Reading tests...")
+if args.debug:
```
Comment (jwise) [inline Utilities/runtests.py:30]: This means you have to launch the load_tests in a debugger too, right?  Maybe --debug_load_tests should be its own argument for the rare case that you want to do that?

File: rcore/api_func_symbols.h [hunk 2]
```diff
@@ -60,28 +61,7 @@ UNIMPL(_clock_copy_time_string);
...
-UNIMPL(_dict_write_data);
-UNIMPL(_dict_write_end);
-UNIMPL(_dict_write_int);
-UNIMPL(_dict_write_int16);
-UNIMPL(_dict_write_int32);
-UNIMPL(_dict_write_int8);
-UNIMPL(_dict_write_tuplet);
-UNIMPL(_dict_write_uint16);
-UNIMPL(_dict_write_uint32);
-UNIMPL(_dict_write_uint8);
```
Comment (jwise) [inline rcore/api_func_symbols.h:84]: 🚀

**Comment by jwise** [comment]
Well this freaking rules.  LGTM.

File: Utilities/runtests.py (hunk 1 above)
Comment (ginge) [reply Utilities/runtests.py]: aye good point. removed the commit for now

**Pull request merged**